### PR TITLE
Product listing

### PR DESCRIPTION
**Pull request by chanchal16**
Created product listing page
- added products page
- added filters
- sort by price
- filter by a category
- filter by price
- filter by ratings

Live link: https://festive-elion-38ef48.netlify.app/

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,12 +1,15 @@
 import "./App.css";
 import Navbar from "./components/Navbar";
```
Comment (sonishreyas) [inline src/App.js:2]: You can add a index.js file and export all components from there so you can import it like this.
```
import {Footer, Navbar,Sidebar} from "./components"
```
This makes the code look more clean

File: src/Utils/filterUtils.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+const getSortedProducts = (products,sortBy)=>{
```
Comment (sonishreyas) [inline src/Utils/filterUtils.js:1]: You can add jsdoc documentation for better understanding of code

File: src/components/Sidebar.jsx [hunk 3]
```diff
@@ -31,24 +35,21 @@ export default function Sidebar() {
...
-                    className="--slider accent" id="myRange" 
-                    onChange={(e) => dispatch({ type: "PRICE", payload: e.target.value })} />*/}
                 </div>
                 {/* ---categories-- */}
                 <div className="filters">
                     <p className="text-md">Category</p>
                     <div className='category-div'>
-                    <label className="text-sm accent" for="checkbox">
-                        <input type="checkbox" name="category" /> All </label> 
+                    {/* <label className="text-sm accent" for="checkbox">
```
Comment (sonishreyas) [inline src/components/Sidebar.jsx:43]: You can remove the commented code

File: src/contexts/filterContext.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    products:[],
+    category:{eyeglasses:false,computerglasses:false,sunglasses:false},
+    rating:null ,
+    range:'' 
+}
+
+const FilterProvider = ({ children }) => {
+ 
+  const [state, dispatch] = useReducer(filterReducer, initialFilterState);
+  console.log("state", state);
```
Comment (sonishreyas) [inline src/contexts/filterContext.jsx:16]: Remoce all the console.log from the code before merging to dev

**Review by sonishreyas — APPROVED** [review]
The code looks clean and easy to understand. I had a few suggestion, you can implement.

File: src/App.js (hunk 1 above)
Comment (chanchal16) [reply src/App.js:2]: sure will do that!

File: src/Utils/filterUtils.js (hunk 2 above)
Comment (chanchal16) [reply src/Utils/filterUtils.js:1]: ok, will include that.

File: src/contexts/filterContext.jsx (hunk 4 above)
Comment (shraddha15-creator) [inline src/contexts/filterContext.jsx:16]: remove console logs

File: src/contexts/filterContext.jsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+//   console.log('produts',products)
```
Comment (shraddha15-creator) [inline src/contexts/filterContext.jsx:17]: remove consoles

**Review by shraddha15-creator — APPROVED** [review]
code looks good. Just one suggestion can we get that ` sort by low-to-high & high-to-low ` in the filter sidebar it will look good all filters at one side. Else everything looks good to go.

**Comment by chanchal16** [comment]
> code looks good. Just one suggestion can we get that `sort by low-to-high & high-to-low` in the filter sidebar it will look good all filters at one side. Else everything looks good to go.

Yes, I was thinking of changing it. 
Thanks for the feedback!!

**Pull request merged**